core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -526,6 +526,9 @@ func (vdb *VersionedDB) readFromDB(namespace, key string) (*keyValue, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err := validateKey(key); err != nil {
+		return nil, err
+	}
```

Comment:
If the goal is to avoid sending invalid keys to CouchDB but not to the cache, this PR LGTM. Is that the goal? If not, we need to perform this check before accessing the cache also.

Reply:
Cache would eventually be populated from DB content only. So, checking in cache does not add any value. Rather, in the current way, it amortize the cost the cost of this check.